File: src/main/java/com/emeraldElves/alcohollabelproject/Data/AlcoholInfo.java
package com.emeraldElves.alcohollabelproject.Data;

import java.util.Objects;

public class AlcoholInfo {

    public static final String DB_TABLE = "alcohol";
    public static final String DB_BRAND_NAME = "brand_name";
    public static final String DB_ALCOHOL_TYPE = "alcohol_type";
    public static final String DB_SERIAL_NUMBER = "serial_number";
    public static final String DB_ORIGIN = "origin";
    public static final String DB_ALCOHOL_CONTENT = "alcohol_content";
    public static final String DB_FANCICUL_NAME = "fanciful_name";
    public static final String DB_ID = "id";

    private String brandName;
    private AlcoholType type;
    private String serialNumber;
    private ProductSource origin;

    private long id;
    private String fancifulName;
    private double alcoholContent;
    private String formula;

    public AlcoholInfo(String brandName, AlcoholType type, String serialNumber, ProductSource origin) {
        this.brandName = brandName;
        this.type = type;
        this.serialNumber = serialNumber;
        this.origin = origin;
        id = -1;
        fancifulName = "";
        alcoholContent = 0;
    }

    public String getBrandName() {
        return brandName;
    }

    public AlcoholType getType() {
        return type;
    }

    public String getSerialNumber() {
        return serialNumber;
    }

    public ProductSource getOrigin() {
        return origin;
    }

    public long getId() {
        return id;
    }

    public String getFancifulName() {
        return fancifulName;
    }

    public double getAlcoholContent() {
        return alcoholContent;
    }

    public String getFormula() {
        return formula;
    }

    public void setFormula(String formula) {
        this.formula = formula;
    }

    public void setId(long id) {
        this.id = id;
    }

    public void setFancifulName(String fancifulName) {
        this.fancifulName = fancifulName;
    }

    public void setAlcoholContent(double alcoholContent) {
        this.alcoholContent = alcoholContent;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        AlcoholInfo that = (AlcoholInfo) o;
        return id == that.id &&
                Double.compare(that.alcoholContent, alcoholContent) == 0 &&
                Objects.equals(brandName, that.brandName) &&
                type == that.type &&
                Objects.equals(serialNumber, that.serialNumber) &&
                origin == that.origin &&
                Objects.equals(fancifulName, that.fancifulName);
    }

    @Override
    public int hashCode() {

        return Objects.hash(brandName, type, serialNumber, origin, id, fancifulName, alcoholContent);
    }
}

File: src/main/java/com/emeraldElves/alcohollabelproject/Data/AlcoholInfo.kt
package com.emeraldElves.alcohollabelproject.Data

/**
 * Created by Kyle on 5/30/2017.
 */
//open class AlcoholInfo(var alcoholContent: Double, val fancifulName: String, val brandName: String,
//                       val origin: ProductSource, val type: AlcoholType, val serialNumber: String,
//                       val formula: String)

open class AlcoholInfo(val brandName: String, val type: AlcoholType, val serialNumber: String, val origin: ProductSource){
    var fancifulName: String = ""
    var formula: String = ""
    var alcoholContent: Double = 0.0
    var id: Long = -1

    companion object {
        val DB_TABLE = "alcohol"
        val DB_BRAND_NAME = "brand_name"
        val DB_ALCOHOL_TYPE = "alcohol_type"
        val DB_SERIAL_NUMBER = "serial_number"
        val DB_ORIGIN = "origin"
        val DB_ALCOHOL_CONTENT = "alcohol_content"
        val DB_FANCICUL_NAME = "fanciful_name"
        val DB_ID = "id"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as AlcoholInfo

        if (brandName != other.brandName) return false
        if (type != other.type) return false
        if (serialNumber != other.serialNumber) return false
        if (origin != other.origin) return false
        if (fancifulName != other.fancifulName) return false
        if (formula != other.formula) return false
        if (alcoholContent != other.alcoholContent) return false
        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        var result = brandName.hashCode()
        result = 31 * result + type.hashCode()
        result = 31 * result + serialNumber.hashCode()
        result = 31 * result + origin.hashCode()
        result = 31 * result + fancifulName.hashCode()
        result = 31 * result + formula.hashCode()
        result = 31 * result + alcoholContent.hashCode()
        result = 31 * result + id.hashCode()
        return result
    }


}

//class WineInfo(alcoholContent: Double, fancifulName: String, brandName: String, origin: ProductSource, serialNumber: String, formula: String, var pH: Double, var vintageYear: Int, var grapeVarietal: String, var appellation: String) : AlcoholInfo(alcoholContent, fancifulName, brandName, origin, AlcoholType.WINE, serialNumber, formula)
//
//class BeerInfo(alcoholContent: Double, fancifulName: String, brandName: String, origin: ProductSource, serialNumber: String, formula: String) : AlcoholInfo(alcoholContent, fancifulName, brandName, origin, AlcoholType.BEER, serialNumber, formula)
//
//class SpiritInfo(alcoholContent: Double, fancifulName: String, brandName: String, origin: ProductSource, serialNumber: String, formula: String) : AlcoholInfo(alcoholContent, fancifulName, brandName, origin, AlcoholType.DISTILLEDSPIRITS, serialNumber, formula)